.github/workflows/golangci-lint.yml

```diff
@@ -1,20 +1,23 @@
 name: golangci-lint
 on:
+  pull_request:
   push:
     tags:
       - v*
     branches:
-      - master
       - main
-  pull_request:
+
 jobs:
   golangci:
     name: lint
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/setup-go@v2
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
+
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@v2
+        uses: golangci/golangci-lint-action@v3
         with:
-          version: v1.45.2
+          version: v1.45
```

Comment:
why the downgrade to the minor version vs specifying the patch version too?

Reply:
Auto updates the patch. Ideally I think we'd want latest, but its not an option. In the future we'll want all of our repos to kinda be the same without having to maintain each patch version 😁